client/src/components/layout/ParkShow.js

```diff
@@ -1,38 +1,85 @@
-import React, {useEffect, useState} from "react"
+import React, { useEffect, useState } from "react";
+import NewReviewForm from "../NewReviewForm.js";
+import ReviewTile from "./ReviewTile.js";
 
-const ParkShow = props => {
-  const [park, setPark] = useState({})
+const ParkShow = (props) => {
+  const [park, setPark] = useState({
+    name: "",
+    picture: "",
+    location: "",
+    description: "",
+    rating: "",
+    reviews: [],
+  });
 
   const getPark = async () => {
     const parkId = props.match.params.id;
-
     try {
       const response = await fetch(`/api/v1/parks/${parkId}`);
-      if(!response.ok) {
-        const errorMessage = `${response.status} (${response.statusText})`
-        const error = new Error(errorMessage)
-        throw(error)
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`;
+        const error = new Error(errorMessage);
+        throw error;
       }
       const body = await response.json();
       setPark(body.park);
     } catch (error) {
-      console.error(`Err in fetch: ${error.message}`)
+      console.error(`Err in fetch: ${error.message}`);
+    }
+  };
+
+  const postReview = async (newReviewData) => {
+    try {
+      const parkId = props.match.params.id;
+      const response = await fetch(`/api/v1/parks/${parkId}/reviews`, {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json",
+        }),
+        body: JSON.stringify(newReviewData),
+      });
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json();
+          const newErrors = translateServerErrors(body.errors);
+          return setErrors(newErrors);
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`;
+          const error = new Error(errorMessage);
+          throw error;
+        }
+      } else {
+        const body = await response.json();
+        setPark({
+          ...park,
+          reviews: [...park.reviews, body.review],
+        });
+        getPark();
```

Comment:
nooooooooo and the code was so nice! Why are we getting the park again? that info hasnt changed. remove line 57 and this should work the same, unless there is someting I am missing?